### ISSUE TITLE
Not properly handling "Project up to date"

### ISSUE THREAD

**Issue by DynConcepts** [mmajcica/DevEnvBuild#16]
Experimentation with direct invocation of DevEnv.com seems to indicate that VcVars32.bat must be invoked. If that happens (I use the simply CmdTask) then things work, without it (Still CmdTask) they do not..... Clearly the Task here will not....

The killer is that getting this functionality is for speed (yes, seconds count) and invocation fo VcVars32 for hundreds or thousands of solutions (yes, I am serious) it really adds up.

**Comment by mmajcica** [comment]
I am not familiar with that file, judging by the name and a quick google search seems it's used in vc builds.
Also, seems that with the more recent versions of the VS, getting the path of the VcVars32.bat is not straight forward.
How do you invoke it? before calling the deveenv.com in order to populate some session variables?
If you are willing to help with sorting out the path and testing, I can quickly add the support for it.

**Comment by DynConcepts** [comment]
Thanks for the response during the holidays 👍  (I have a presentation on Tuesday the 5th which will be impacted (moderate) with this...

I do not think the VcVars needs to necessarily be called, but rather there is some setting(s) such as PAth, Env, ??? that influences exactly how devenv.com does its processing...

Here are the two relevant task definitions... One can see the difference by swapping the two (do *not* run both in the same build, that will never produce and uptodate by either.


```
    - task: CmdLine@2
      displayName:  (DevEnv via CmdLine Task) ServerContextSupport - TypeLib/Win32 
      inputs:
        script: |
          "C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\VC\Auxiliary\Build\vcVars32.bat"
          Echo On
          "C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\Common7\IDE\devenv.com" /Build "TypeLib|Win32" ".\ServerContextSupport\Src\ServerContextSupport.sln"
  
#    - task: DevEnvBuild@2
#      displayName:  (DevEnv via Marketplace Task) ServerContextSupport - TypeLib/Win32 
#      inputs:
#        solution: '.\ServerContextSupport\Src\ServerContextSupport.sln'
#        buildCfg: 'TypeLib'
#        buildPlatform: 'Win32'
#        vsVersion: '15.0'


```

Catch you after the weekend (I will be working all weekend, so if there is an earlier response, I will be here to react to it.

**Comment by mmajcica** [comment]
I did some more research and my plan is to do the following.
Add an option in the task that will state

```
{
      "name": "vcvars",
      "type": "boolean",
      "label": "Include VcVars",
      "defaultValue": "false",
      "required": false,
      "helpMarkDown": "Set environment variables to configure the command line for native 32-bit or 64-bit compilation."
}
```
Once selected, you will get to choose the params for the invocation of VsDevCmd.bat as stated here
[https://docs.microsoft.com/en-us/cpp/build/building-on-the-command-line?view=msvc-160#developer_command_file_locations](https://docs.microsoft.com/en-us/cpp/build/building-on-the-command-line?view=msvc-160#developer_command_file_locations)

So, if the option is selected, and based on chosen params, before invoking the devenv, VsDevCmd will be called.

Would that work?

Also, by reading that document I found out the following:
Starting in Visual Studio version 16.5, DEVENV does not use the command-line environment to control the toolset and libraries used.

Should we account for that somehow?
